imap/imap.c

```diff
@@ -234,17 +303,28 @@ static int sync_helper(struct Mailbox *m, AclFlags right, enum MessageType flag,
   if ((right == MUTT_ACL_WRITE) && !imap_has_flag(&imap_mdata_get(m)->flags, name))
     return 0;
 
+  int count = 0;
+  char buf[1024] = { 0 };
+
+  struct UidArray uida = ARRAY_HEAD_INITIALIZER;
+
+  // Set the flag (+FLAGS) on matching emails
+  select_email_uids(emails, num_emails, flag, true, false, &uida);
```

Comment:
Don't we need to sort the array here? Also, would it make sense to move the sorting inside `select_email_uids` and `emails_to_uid_array`?

Reply:
> Don't we need to sort the array here?

No, because we sorted `emails` in `imap_sync_mailbox()`.

https://github.com/neomutt/neomutt/blob/76ba1c95423b760a120e6c3afc85c1e6db1c8a65/imap/imap.c#L1565-L1568

> would it make sense to move the sorting inside `select_email_uids()`

Ideally, yes, but `imap_sync_mailbox()` leads to **ten** `select_email_uids()` calls in quick succession.
Using the same set of emails, it calls `sync_helper()` for each attribute.

https://github.com/neomutt/neomutt/blob/76ba1c95423b760a120e6c3afc85c1e6db1c8a65/imap/imap.c#L1570-L1578

and `sync_helper()` calls `select_email_uids()` and `imap_exec_msg_set()` twice:

https://github.com/neomutt/neomutt/blob/76ba1c95423b760a120e6c3afc85c1e6db1c8a65/imap/imap.c#L311-L327

> and `emails_to_uid_array()`

Yes, that's a bit tidier. Done.